src/env.cc

```diff
@@ -153,4 +153,30 @@ void Environment::PrintSyncTrace() const {
   fflush(stderr);
 }
 
+struct Environment::AtExitCallback {
+  AtExitCallback* next_;
+  void (*cb_)(void* arg);
+  void* arg_;
+};
+
+void Environment::RunAtExitCallbacks() {
+  AtExitCallback* p = at_exit_functions_;
+  at_exit_functions_ = nullptr;
+
+  while (p) {
```

Comment:
Absolutely, but perhaps I can land https://github.com/nodejs/node/pull/12255 and rebase this and this code would not be used any longer.